scrapy/squeues.py

```diff
@@ -8,21 +8,23 @@
 
 from queuelib import queue
 
+from scrapy.exceptions import SerializationError
 from scrapy.utils.reqser import request_to_dict, request_from_dict
 
 
 def _with_mkdir(queue_class):
 
-    class DirectoriesCreated(queue_class):
+    class DirectoriesCreatedForKey(queue_class):
 
-        def __init__(self, path, *args, **kwargs):
+        def __init__(self, _, key, *args, **kwargs):
```

Comment:
nitpick: for me the could would be easier to read if `crawler` name is used instead of `_`. It took me some time to figure out what is this `_` which is passed through.